### ISSUE TITLE
Warning from test_collections.test_picking

### ISSUE THREAD

**Issue by mdboom** [matplotlib/matplotlib#5362]
```
matplotlib.tests.test_collections.test_picking ... /matplotlib/collections.py:360: UserWarning: Collection picker None could not be converted to float
  % self._picker)
ok
```

**Comment by tacaswell** [comment]
I have dug into this one before and it is a mess.  The attribute in question can either be a function or a float and iirc the warning is reasonable.   This is one thing that with traitlets we require it to be a function and a float is assigned make a properly curried default function.

**Comment by anntzer** [comment]
Should have been closed by #6491.
